### PR TITLE
Hardening the list against hard stops (ie. stopping scrolling by taping the screen).

### LINKED ISSUE #76: populateItemDetail() is never called after scrollInstantly()
After running `.scrollInstantly()` fast-list is flagged as `!idle` so `.populateItemDetail()` is never run. This is also the case when scrolling fast and then tapping the list to halt scrolling.

It seems we need to implement some logic that always runs `.popuateItemDetail()` once scrolling has finished.

Any suggestions @etiennesegonzac?

### PR DESCRIPTION
**Pull request by etiennesegonzac**
Making sure we always fully render in this case.

Closes #76

### CODE REVIEW

File: fast-list.js [hunk 1]
```diff
@@ -60,6 +60,7 @@ function FastList(source) {
   this.geometry.itemHeight = source.getItemHeight();
 
   this._rendered = false;
+  this._scrollSentinel = null;
```
Comment (wilsonpage) [inline fast-list.js]: What does 'scroll sentinel' mean?

File: fast-list.js [hunk 2]
```diff
@@ -389,10 +390,21 @@ FastList.prototype = {
...
     else this.render();
+
+    if (this.geometry.idle) return;
+
+    // We just did a partial rendering and need to make sure it won't
+    // get stuck this way if the scrolling comes to a hard stop.
+    this._scrollSentinel = setTimeout(() => {
+      this.updateRenderingWindow(true);
+      this.render();
+    }, 200);
```
Comment (wilsonpage) [inline fast-list.js:407]: This seems like a potentially a bad thing to have in such a hot function?

File: fast-list.js [hunk 3]
```diff
@@ -389,10 +390,21 @@ FastList.prototype = {
    */
 
   handleScroll: function(evt) {
+    window.clearTimeout(this._scrollSentinel);
```
Comment (wilsonpage) [inline fast-list.js]: NIT: `clearTimeout(this._scrollSentinel);`

File: fast-list.js (hunk 2 above)
Comment (wilsonpage) [inline fast-list.js:407]: I'm sure you've done your performance checks though :)

Comment (etiennesegonzac) [inline fast-list.js:407]: Well, the `200` duration was pretty carefully chosen yes.
Not sure if the constant clearTimeout/seTimeout would show up in profiles, but I'm not sure how we can do without it.

Good news is bug 1172171 has an assignee :)

File: fast-list.js (hunk 1 above)
Comment (etiennesegonzac) [inline fast-list.js]: would `scrollStopSentinel` be clearer? I'm really open to suggestions :)

File: fast-list.js (hunk 2 above)
Comment (wilsonpage) [inline fast-list.js:407]: Wow, that would be really helpful!

File: fast-list.js (hunk 1 above)
Comment (wilsonpage) [inline fast-list.js]: `scrollStopTimeout` makes the most sense to me :)

**Comment by etiennesegonzac** [comment]
`_scrollStopTimeout` it is!

**Comment by etiennesegonzac** [comment]
I'll give this PR a profiler check tomorrow before merging (it the check looks good :))

**Comment by etiennesegonzac** [comment]
Just confirmed that:
- the constant clearTimeout/setTimeout has no visible impact on the `handleScroll` execution time
- we don't trigger false rendering while scrolling (the timeout duration is well tuned)
  Merging!

**Pull request merged**